common/Environments/Helpers/ComputeSystemHelpers.cs

```diff
@@ -140,19 +141,32 @@ public static EnvironmentsCallToActionData UpdateCallToActionText(int providerCo
     /// this method is used to remove all items individually from the end of the collection to the beginning of the collection.
     /// </remarks>
     /// <typeparam name="T">Type of objects that the collection contains</typeparam>
-    /// <param name="collection">An observable collection that contains zero to N elements</param>
-    public static void RemoveAllItems<T>(ObservableCollection<T> collection)
+    /// <param name="collectionToUpdate">An observable collection that contains zero to N elements that will have its contents replaced</param>
+    /// <param name="listWithUpdates">A list that contains zero to N elements whose elements will be added to collectionToUpdate</param>
+    /// <returns>
+    /// True only if we successfully replaced all items in the collection. False otherwise.
+    /// </returns>
+    public static bool RemoveAllItemsAnReplace<T>(ObservableCollection<T> collectionToUpdate, List<T> listWithUpdates)
     {
         try
         {
-            for (var i = collection.Count - 1; i >= 0; i--)
+            for (var i = collectionToUpdate.Count - 1; i >= 0; i--)
             {
-                collection.RemoveAt(i);
+                collectionToUpdate.RemoveAt(i);
             }
+
+            for (var i = 0; i < listWithUpdates.Count; i++)
+            {
+                collectionToUpdate.Add(listWithUpdates[i]);
+            }
+

```

Comment:
These loops fire OnCollectionChanged events for every operation which is inefficient and can make UX less responsive.
There is [BulkObservableCollection](https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.language.intellisense.bulkobservablecollection-1?view=visualstudiosdk-2022) class that has [AddRange](https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.language.intellisense.bulkobservablecollection-1.addrange?view=visualstudiosdk-2022#microsoft-visualstudio-language-intellisense-bulkobservablecollection-1-addrange(system-collections-generic-ienumerable((-0)))) method and 
[BeginBulkOperation](https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.language.intellisense.bulkobservablecollection-1.beginbulkoperation?view=visualstudiosdk-2022#microsoft-visualstudio-language-intellisense-bulkobservablecollection-1-beginbulkoperation) and [EndBulkOperation](https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.language.intellisense.bulkobservablecollection-1.endbulkoperation?view=visualstudiosdk-2022#microsoft-visualstudio-language-intellisense-bulkobservablecollection-1-endbulkoperation) methods that can be used to suspend notification while doing multiple changes to the collection.

Reply:
Using that requires adding the `Microsoft.VisualStudio.Language.Intellisense` nuget package so I don't think we want to add it this close to cut off. Luckily the amount of properties sent back by both the Hyper-V extension and Dev Box are small (4 for dev box and 5 for hyper-V). So I can investigate using the above nuget package after build